src/protocol/prf_sharding/mod.rs

```diff
@@ -252,6 +266,138 @@ where
     rows_chunked_by_user
 }
 
+/// # Errors
+/// If there is an issue in multiplication, it will error
+pub async fn attribution_and_capping_and_aggregation<C, BK, TV, F, S, SB>(
+    sh_ctx: C,
+    input_rows: Vec<PrfShardedIpaInputRow<BK, TV>>,
+    num_saturating_sum_bits: usize,
+) -> Result<Vec<S>, Error>
+where
+    C: UpgradableContext,
+    C::UpgradedContext<F>: UpgradedContext<F, Share = S>,
+    S: LinearSecretSharing<F> + Serializable + SecureMul<C::UpgradedContext<F>>,
+    for<'a> &'a S: LinearRefOps<'a, S, F>,
+    C::UpgradedContext<Gf2>: UpgradedContext<Gf2, Share = Replicated<Gf2>>,
+    F: PrimeField + ExtendableField,
+    TV: GaloisField,
+    BK: GaloisField,
+{
+    // Call attribution_and_capping
+    let user_level_attributions: Vec<CappedAttributionOutputs> =
+        attribution_and_capping(sh_ctx.clone(), input_rows, num_saturating_sum_bits).await?;
+
+    let prime_field_validator = sh_ctx.narrow(&Step::BinaryValidator).validator::<F>();
```

Comment:
If you move lines 290-291 up to the top of this method, you can probably avoid the need to "clone" one line 288.